File: src/logistic_regression.R
training.data.raw <- read.csv('../input/titanic_data.csv',header=T,na.strings=c(""))
#drop cabin etc as there are lot of missing values.
data <- subset(training.data.raw,select=c(2,3,5,6,7,8,10,12))
data$Age[is.na(data$Age)] <- mean(data$Age,na.rm=T)
data <- data[!is.na(data$Embarked),]
train <- data[1:800,]
test <- data[801:889,]

model <- glm(Survived ~.,family=binomial(link='logit'),data=train)

fitted.results <- predict(model,newdata=subset(test,select=c(2,3,4,5,6,7,8)),type='response')
fitted.results <- ifelse(fitted.results > 0.5,1,0)

misClasificError <- mean(fitted.results != test$Survived)
print(paste('Accuracy',1-misClasificError))
#should say something like [1] "Accuracy 0.842696629213483"

File: src/logistic_regression.py
import pandas as pd
import numpy as np
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split

train = pd.read_csv('../input/titanic_data.csv');
corrmat = train.corr();

train.info();

def normalize(inputData):
    temp = inputData.fillna(value = inputData.mean())
    mu = inputData.mean()
    sigma = inputData.std()
    return (temp - mu) / sigma

def denormalize(inputData, mu, sigma):
    return (inputData * sigma) + mu

def sigmoid(z):
    return  1.0 / ( 1 + np.exp(-z))


train.Sex = train.Sex.astype('category')
train.Embarked = train.Embarked.fillna(value = 'S')
train.Embarked = train.Embarked.astype('category')
train = train.drop(['Name','Ticket','Cabin'],axis=1)
train.set_index('PassengerId');
train.info()
mean_age = train.Age.mean()
train.Age = train.Age.fillna(mean_age)
train.Age = normalize(train.Age)
train.Fare = normalize(train.Fare)
cols_to_transform = [ 'Sex','Embarked' ]
train = pd.get_dummies(train, columns = cols_to_transform )

y = train['Survived']
X = train[['Pclass', 'Age','SibSp','Parch','Fare','Sex_female','Sex_male','Embarked_C','Embarked_Q','Embarked_S']]

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=21)

logit = LogisticRegression(max_iter=400)
logit.fit(X_train,y_train)

pred = logit.predict(X_test)
print("accuracy %f"%(float(sum(abs(y_test-pred == 0)))/len(pred)))